packages/components/src/modal/index.tsx

```diff
@@ -209,7 +209,7 @@ function UnforwardedModal(
 
 		if (
 			shouldCloseOnEsc &&
-			event.code === 'Escape' &&
+			( event.code === 'Escape' || event.key === 'Escape' ) &&
```

Comment:
`@ariakit/test` keyboard events use `event.key`, so I had to add this check in order for the `Modal` component to react to those simulated user events

Reply:
We can implement `code` in `@ariakit/test`. I'm just not sure how to transform keys into code, and I'd prefer to sidestep the need to maintain a full map of values.

Reply:
I guess we can keep this change to the `Modal` component for now (cc @mirka )